core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -520,7 +521,13 @@ sealed abstract class IO[+E, +A] { self =>
   /**
    * Runs this action in a new fiber, resuming when the fiber terminates.
    */
-  final def run[E1 >: E, A1 >: A]: IO[Nothing, ExitResult[E1, A1]] = new IO.Run(self)
+  final def run: IO[Nothing, ExitResult[E, A]] =
```

Comment:
It's nice to see this go away as a primitive. Also in this form, it becomes simple to supervise the `fork` so that if this IO is interrupted, then the child fiber will be interrupted. We can think about whether or not to do that.